api/src/main/java/org/openmrs/module/registrationcore/api/impl/IdentifierBuilder.java

```diff
@@ -37,14 +37,26 @@ public void setPatientService(PatientService patientService) {
      * @param location identifier location
      * @return generated patient identifier
      */
-    public PatientIdentifier generateIdentifier(Integer sourceId, Location location) {
+	public PatientIdentifier generateIdentifier(Integer sourceId, Location location) {
+		return generateIdentifier((Object) sourceId, location);
+	}
+	
+	/**
+	 * Generate new identifier.
+	 * 
+	 * @param sourceId identifier source uuid or id
+	 * @param location identifier location
+	 * @return generated patient identifier
+	 */
+	public PatientIdentifier generateIdentifier(Object sourceId, Location location) {
```

Comment:
I don't love this.  Having 2 methods with the same name/signature, where one takes in the same arguments as the other is very confusing.  Can you name the method differently so there is no chance of confusion as to which method is hit if you pass in an Integer, Location pair?